spyder/plugins/ipythonconsole.py

```diff
@@ -590,7 +590,7 @@ class IPythonConsole(SpyderPluginWidget):
     focus_changed = Signal()
     edit_goto = Signal((str, int, str), (str, int, str, bool))
 
-    def __init__(self, parent, testing=False):
+    def __init__(self, parent, testing=False, test_dir=programs.TEMPDIR):

```

Comment:
Please use here only `TEMPDIR`, so import it from `programs` with

    from programs import TEMPDIR

Reply:
We also need to add

    self.test_dir = test_dir